### ISSUE TITLE
SAML's authentication should not block transport_worker threads

### ISSUE THREAD

**Issue by slobodanadamovic** [elastic/elasticsearch#104962]
### Description

SAML's metadata can be configured to be loaded from a file or an external HTTP URL. The fetching and refreshing of metadata is done periodically in a dedicated timer thread. But since the metadata can be changed asynchronously, it can cause all authentications to fail until the periodic refresh kicks in. Hence, the SAML realm tries to leniently fetch and refresh metadata during failed authentication. This refreshing will be executed on the same thread that is executing the authentication. Which appears to be the `transport_worker` thread. Blocking [networking threads](https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-network.html#modules-network-threading-model) can cause the cluster instability due to delay of message processing and eventually lead to a node [failing its health checks](https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-fault-detection.html) and its removal from the cluster.

The loading metadata from a URL is particularly bad since it can fail/be slow due to many factors (DNS resolution, slow or unreachable metadata server, etc). 

We may want to evaluate the proposal in https://github.com/elastic/elasticsearch/issues/68004 to make the Security async. But in the mean time we should make sure that already identified blocking SAML code is executed async and not on a `transport_worker` threads. We can do this by making the SAML authentication calls asynchronous by forking a dedicated thread - like we do for [LDAP realm authentication](https://github.com/elastic/elasticsearch/blob/35cc9e115916731f64efb141b857167544858e3c/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ldap/LdapRealm.java#L149-L150).

<details>
<summary> Thread dump which shows two transport_worker threads being blocked on a lock obtained by timer thread</summary>

```
"Timer for org.elasticsearch.xpack.security.authc.saml.SamlRealm$PrivilegedHTTPMetadataResolver@43317844" #58 [297] daemon prio=5 os_prio=0 cpu=5322.34ms elapsed=2312956.01s tid=0x00007eff643ddb80 nid=297 runnable  [0x00007f004c371000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.SocketDispatcher.read0(java.base@20.0.2/Native Method)
	at sun.nio.ch.SocketDispatcher.read(java.base@20.0.2/SocketDispatcher.java:47)
	at sun.nio.ch.NioSocketImpl.tryRead(java.base@20.0.2/NioSocketImpl.java:256)
	at sun.nio.ch.NioSocketImpl.implRead(java.base@20.0.2/NioSocketImpl.java:307)
	at sun.nio.ch.NioSocketImpl.read(java.base@20.0.2/NioSocketImpl.java:345)
	at sun.nio.ch.NioSocketImpl$1.read(java.base@20.0.2/NioSocketImpl.java:794)
	at java.net.Socket$SocketInputStream.read(java.base@20.0.2/Socket.java:1025)
	at sun.security.ssl.SSLSocketInputRecord.read(java.base@20.0.2/SSLSocketInputRecord.java:483)
	at sun.security.ssl.SSLSocketInputRecord.readFully(java.base@20.0.2/SSLSocketInputRecord.java:466)
	at sun.security.ssl.SSLSocketInputRecord.decodeInputRecord(java.base@20.0.2/SSLSocketInputRecord.java:243)
	at sun.security.ssl.SSLSocketInputRecord.decode(java.base@20.0.2/SSLSocketInputRecord.java:181)
	at sun.security.ssl.SSLTransport.decode(java.base@20.0.2/SSLTransport.java:111)
	at sun.security.ssl.SSLSocketImpl.decode(java.base@20.0.2/SSLSocketImpl.java:1509)
	at sun.security.ssl.SSLSocketImpl.readApplicationRecord(java.base@20.0.2/SSLSocketImpl.java:1480)
	at sun.security.ssl.SSLSocketImpl$AppInputStream.read(java.base@20.0.2/SSLSocketImpl.java:1066)
	at org.apache.http.impl.io.SessionInputBufferImpl.streamRead(org.apache.httpcomponents.httpcore@4.4.13/SessionInputBufferImpl.java:137)
	at org.apache.http.impl.io.SessionInputBufferImpl.fillBuffer(org.apache.httpcomponents.httpcore@4.4.13/SessionInputBufferImpl.java:153)
	at org.apache.http.impl.io.SessionInputBufferImpl.readLine(org.apache.httpcomponents.httpcore@4.4.13/SessionInputBufferImpl.java:280)
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(org.apache.httpcomponents.httpclient@4.5.13/DefaultHttpResponseParser.java:138)
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(org.apache.httpcomponents.httpclient@4.5.13/DefaultHttpResponseParser.java:56)
	at org.apache.http.impl.io.AbstractMessageParser.parse(org.apache.httpcomponents.httpcore@4.4.13/AbstractMessageParser.java:259)
	at org.apache.http.impl.DefaultBHttpClientConnection.receiveResponseHeader(org.apache.httpcomponents.httpcore@4.4.13/DefaultBHttpClientConnection.java:163)
	at org.apache.http.impl.conn.CPoolProxy.receiveResponseHeader(org.apache.httpcomponents.httpclient@4.5.13/CPoolProxy.java:157)
	at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(org.apache.httpcomponents.httpcore@4.4.13/HttpRequestExecutor.java:273)
	at org.apache.http.protocol.HttpRequestExecutor.execute(org.apache.httpcomponents.httpcore@4.4.13/HttpRequestExecutor.java:125)
	at org.apache.http.impl.execchain.MainClientExec.execute(org.apache.httpcomponents.httpclient@4.5.13/MainClientExec.java:272)
	at org.apache.http.impl.execchain.ProtocolExec.execute(org.apache.httpcomponents.httpclient@4.5.13/ProtocolExec.java:186)
	at org.apache.http.impl.execchain.RetryExec.execute(org.apache.httpcomponents.httpclient@4.5.13/RetryExec.java:89)
	at org.apache.http.impl.execchain.RedirectExec.execute(org.apache.httpcomponents.httpclient@4.5.13/RedirectExec.java:110)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(org.apache.httpcomponents.httpclient@4.5.13/InternalHttpClient.java:185)
	at org.apache.http.impl.client.CloseableHttpClient.execute(org.apache.httpcomponents.httpclient@4.5.13/CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(org.apache.httpcomponents.httpclient@4.5.13/CloseableHttpClient.java:56)
	at org.opensaml.saml.metadata.resolver.impl.HTTPMetadataResolver.fetchMetadata(org.opensaml.saml.impl@4.3.0/HTTPMetadataResolver.java:212)
	at org.elasticsearch.xpack.security.authc.saml.SamlRealm$PrivilegedHTTPMetadataResolver.access$001(org.elasticsearch.security@8.10.2/SamlRealm.java:741)
	at org.elasticsearch.xpack.security.authc.saml.SamlRealm$PrivilegedHTTPMetadataResolver.lambda$fetchMetadata$0(org.elasticsearch.security@8.10.2/SamlRealm.java:751)
	at org.elasticsearch.xpack.security.authc.saml.SamlRealm$PrivilegedHTTPMetadataResolver$$Lambda$4128/0x00007effe8ea32c0.run(org.elasticsearch.security@8.10.2/Unknown Source)
	at java.security.AccessController.executePrivileged(java.base@20.0.2/AccessController.java:809)
	at java.security.AccessController.doPrivileged(java.base@20.0.2/AccessController.java:571)
	at org.elasticsearch.xpack.security.authc.saml.SamlRealm$PrivilegedHTTPMetadataResolver.fetchMetadata(org.elasticsearch.security@8.10.2/SamlRealm.java:750)
	at org.opensaml.saml.metadata.resolver.impl.AbstractReloadingMetadataResolver.refresh(org.opensaml.saml.impl@4.3.0/AbstractReloadingMetadataResolver.java:364)
	- locked <0x000000062f76c4b8> (a org.elasticsearch.xpack.security.authc.saml.SamlRealm$PrivilegedHTTPMetadataResolver)
	at org.opensaml.saml.metadata.resolver.impl.AbstractReloadingMetadataResolver$RefreshMetadataTask.run(org.opensaml.saml.impl@4.3.0/AbstractReloadingMetadataResolver.java:685)
	at java.util.TimerThread.mainLoop(java.base@20.0.2/Timer.java:566)
	at java.util.TimerThread.run(java.base@20.0.2/Timer.java:516)


"elasticsearch[instance-0000000005][transport_worker][T#3]" #78 [324] daemon prio=5 os_prio=0 cpu=9112936.99ms elapsed=2312947.50s tid=0x00007eff1c018bd0 nid=324 waiting for monitor entry  [0x00007effe29f6000]
   java.lang.Thread.State: BLOCKED (on object monitor)
	at net.shibboleth.utilities.java.support.component.AbstractInitializableComponent.isInitialized(net.shibboleth.utilities.java.support@8.4.0/AbstractInitializableComponent.java:43)
	- waiting to lock <0x000000062f76c4b8> (a org.elasticsearch.xpack.security.authc.saml.SamlRealm$PrivilegedHTTPMetadataResolver)
	at net.shibboleth.utilities.java.support.component.ComponentSupport.ifNotInitializedThrowUninitializedComponentException(net.shibboleth.utilities.java.support@8.4.0/ComponentSupport.java:104)
	at org.opensaml.saml.metadata.resolver.impl.AbstractMetadataResolver.resolveSingle(org.opensaml.saml.impl@4.3.0/AbstractMetadataResolver.java:262)
	at org.elasticsearch.xpack.security.authc.saml.SamlRealm.resolveEntityDescriptorWithPossibleRefresh(org.elasticsearch.security@8.10.2/SamlRealm.java:823)
	at org.elasticsearch.xpack.security.authc.saml.SamlRealm.resolveEntityDescriptor(org.elasticsearch.security@8.10.2/SamlRealm.java:801)
	at org.elasticsearch.xpack.security.authc.saml.SamlRealm.lambda$parseHttpMetadata$9(org.elasticsearch.security@8.10.2/SamlRealm.java:728)
	at org.elasticsearch.xpack.security.authc.saml.SamlRealm$$Lambda$4133/0x00007effe8f25920.run(org.elasticsearch.security@8.10.2/Unknown Source)
	at java.security.AccessController.executePrivileged(java.base@20.0.2/AccessController.java:809)
	at java.security.AccessController.doPrivileged(java.base@20.0.2/AccessController.java:571)
	at org.elasticsearch.xpack.security.authc.saml.SamlRealm.lambda$parseHttpMetadata$10(org.elasticsearch.security@8.10.2/SamlRealm.java:727)
	at org.elasticsearch.xpack.security.authc.saml.SamlRealm$$Lambda$4132/0x00007effe8f25708.get(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.xpack.security.authc.saml.SamlRealm.buildAuthenticationRequest(org.elasticsearch.security@8.10.2/SamlRealm.java:893)
	at org.elasticsearch.xpack.security.action.saml.TransportSamlPrepareAuthenticationAction.prepareAuthentication(org.elasticsearch.security@8.10.2/TransportSamlPrepareAuthenticationAction.java:61)
	at org.elasticsearch.xpack.security.action.saml.TransportSamlPrepareAuthenticationAction.doExecute(org.elasticsearch.security@8.10.2/TransportSamlPrepareAuthenticationAction.java:56)
	at org.elasticsearch.xpack.security.action.saml.TransportSamlPrepareAuthenticationAction.doExecute(org.elasticsearch.security@8.10.2/TransportSamlPrepareAuthenticationAction.java:32)
	at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(org.elasticsearch.server@8.10.2/TransportAction.java:86)
	at org.elasticsearch.action.support.ActionFilter$Simple.apply(org.elasticsearch.server@8.10.2/ActionFilter.java:53)
	at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(org.elasticsearch.server@8.10.2/TransportAction.java:84)
	at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter.lambda$applyInternal$3(org.elasticsearch.security@8.10.2/SecurityActionFilter.java:163)
	at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter$$Lambda$7545/0x00007effe96bf228.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListenerImplementations$DelegatingFailureActionListener.onResponse(org.elasticsearch.server@8.10.2/ActionListenerImplementations.java:212)
	at org.elasticsearch.xpack.security.authz.AuthorizationService.lambda$authorizeAction$8(org.elasticsearch.security@8.10.2/AuthorizationService.java:455)
	at org.elasticsearch.xpack.security.authz.AuthorizationService$$Lambda$7664/0x00007effe96de770.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.xpack.security.authz.AuthorizationService$AuthorizationResultListener.onResponse(org.elasticsearch.security@8.10.2/AuthorizationService.java:1015)
	at org.elasticsearch.xpack.security.authz.AuthorizationService$AuthorizationResultListener.onResponse(org.elasticsearch.security@8.10.2/AuthorizationService.java:981)
	at org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(org.elasticsearch.server@8.10.2/ContextPreservingActionListener.java:32)
	at org.elasticsearch.xpack.security.authz.AuthorizationService.lambda$authorizeAction$9(org.elasticsearch.security@8.10.2/AuthorizationService.java:469)
	at org.elasticsearch.xpack.security.authz.AuthorizationService$$Lambda$7672/0x00007effe96debc0.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(org.elasticsearch.server@8.10.2/ActionListenerImplementations.java:236)
	at org.elasticsearch.xpack.security.authz.RBACEngine.authorizeClusterAction(org.elasticsearch.security@8.10.2/RBACEngine.java:185)
	at org.elasticsearch.xpack.security.authz.AuthorizationService.authorizeAction(org.elasticsearch.security@8.10.2/AuthorizationService.java:459)
	at org.elasticsearch.xpack.security.authz.AuthorizationService.maybeAuthorizeRunAs(org.elasticsearch.security@8.10.2/AuthorizationService.java:435)
	at org.elasticsearch.xpack.security.authz.AuthorizationService.lambda$authorize$3(org.elasticsearch.security@8.10.2/AuthorizationService.java:322)
	at org.elasticsearch.xpack.security.authz.AuthorizationService$$Lambda$7546/0x00007effe96bf450.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListener$2.onResponse(org.elasticsearch.server@8.10.2/ActionListener.java:177)
	at org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(org.elasticsearch.server@8.10.2/ContextPreservingActionListener.java:32)
	at org.elasticsearch.xpack.security.authz.RBACEngine.lambda$resolveAuthorizationInfo$0(org.elasticsearch.security@8.10.2/RBACEngine.java:150)
	at org.elasticsearch.xpack.security.authz.RBACEngine$$Lambda$7548/0x00007effe96bf890.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(org.elasticsearch.server@8.10.2/ActionListenerImplementations.java:236)
	at org.elasticsearch.xpack.security.authz.store.CompositeRolesStore.lambda$getRoles$4(org.elasticsearch.security@8.10.2/CompositeRolesStore.java:194)
	at org.elasticsearch.xpack.security.authz.store.CompositeRolesStore$$Lambda$7549/0x00007effe96bfaa0.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(org.elasticsearch.server@8.10.2/ActionListenerImplementations.java:236)
	at org.elasticsearch.xpack.security.authz.store.CompositeRolesStore.lambda$getRole$5(org.elasticsearch.security@8.10.2/CompositeRolesStore.java:212)
	at org.elasticsearch.xpack.security.authz.store.CompositeRolesStore$$Lambda$8014/0x00007effe9706c98.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(org.elasticsearch.server@8.10.2/ActionListenerImplementations.java:236)
	at org.elasticsearch.xpack.core.security.authz.store.RoleReferenceIntersection.lambda$buildRole$0(org.elasticsearch.xcore@8.10.2/RoleReferenceIntersection.java:49)
	at org.elasticsearch.xpack.core.security.authz.store.RoleReferenceIntersection$$Lambda$8015/0x00007effe9755c40.accept(org.elasticsearch.xcore@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(org.elasticsearch.server@8.10.2/ActionListenerImplementations.java:236)
	at org.elasticsearch.action.support.GroupedActionListener.onResponse(org.elasticsearch.server@8.10.2/GroupedActionListener.java:56)
	at org.elasticsearch.xpack.security.authz.store.CompositeRolesStore.buildRoleFromRoleReference(org.elasticsearch.security@8.10.2/CompositeRolesStore.java:292)
	at org.elasticsearch.xpack.security.authz.store.CompositeRolesStore$$Lambda$8013/0x00007effe9706a70.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.xpack.core.security.authz.store.RoleReferenceIntersection.lambda$buildRole$1(org.elasticsearch.xcore@8.10.2/RoleReferenceIntersection.java:53)
	at org.elasticsearch.xpack.core.security.authz.store.RoleReferenceIntersection$$Lambda$8016/0x00007effe9754800.accept(org.elasticsearch.xcore@8.10.2/Unknown Source)
	at java.lang.Iterable.forEach(java.base@20.0.2/Iterable.java:75)
	at org.elasticsearch.xpack.core.security.authz.store.RoleReferenceIntersection.buildRole(org.elasticsearch.xcore@8.10.2/RoleReferenceIntersection.java:53)
	at org.elasticsearch.xpack.security.authz.store.CompositeRolesStore.getRole(org.elasticsearch.security@8.10.2/CompositeRolesStore.java:210)
	at org.elasticsearch.xpack.security.authz.store.CompositeRolesStore.getRoles(org.elasticsearch.security@8.10.2/CompositeRolesStore.java:187)
	at org.elasticsearch.xpack.security.authz.RBACEngine.resolveAuthorizationInfo(org.elasticsearch.security@8.10.2/RBACEngine.java:146)
	at org.elasticsearch.xpack.security.authz.AuthorizationService.authorize(org.elasticsearch.security@8.10.2/AuthorizationService.java:338)
	at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter.lambda$applyInternal$4(org.elasticsearch.security@8.10.2/SecurityActionFilter.java:159)
	at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter$$Lambda$7544/0x00007effe96bf010.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(org.elasticsearch.server@8.10.2/ActionListenerImplementations.java:236)
	at org.elasticsearch.action.ActionListenerImplementations$MappedActionListener.onResponse(org.elasticsearch.server@8.10.2/ActionListenerImplementations.java:95)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain.authenticateAsync(org.elasticsearch.security@8.10.2/AuthenticatorChain.java:94)
	at org.elasticsearch.xpack.security.authc.AuthenticationService.authenticate(org.elasticsearch.security@8.10.2/AuthenticationService.java:261)
	at org.elasticsearch.xpack.security.authc.AuthenticationService.authenticate(org.elasticsearch.security@8.10.2/AuthenticationService.java:171)
	at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter.applyInternal(org.elasticsearch.security@8.10.2/SecurityActionFilter.java:155)
	at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter.apply(org.elasticsearch.security@8.10.2/SecurityActionFilter.java:114)
	at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(org.elasticsearch.server@8.10.2/TransportAction.java:84)
	at org.elasticsearch.action.support.TransportAction.execute(org.elasticsearch.server@8.10.2/TransportAction.java:61)
	at org.elasticsearch.tasks.TaskManager.registerAndExecute(org.elasticsearch.server@8.10.2/TaskManager.java:199)
	at org.elasticsearch.client.internal.node.NodeClient.executeLocally(org.elasticsearch.server@8.10.2/NodeClient.java:113)
	at org.elasticsearch.client.internal.node.NodeClient.doExecute(org.elasticsearch.server@8.10.2/NodeClient.java:91)
	at org.elasticsearch.client.internal.support.AbstractClient.execute(org.elasticsearch.server@8.10.2/AbstractClient.java:379)
	at org.elasticsearch.xpack.security.rest.action.saml.RestSamlPrepareAuthenticationAction.lambda$innerPrepareRequest$0(org.elasticsearch.security@8.10.2/RestSamlPrepareAuthenticationAction.java:72)
	at org.elasticsearch.xpack.security.rest.action.saml.RestSamlPrepareAuthenticationAction$$Lambda$11408/0x00007effe9b4fc20.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.rest.BaseRestHandler.handleRequest(org.elasticsearch.server@8.10.2/BaseRestHandler.java:103)
	at org.elasticsearch.xpack.security.rest.SecurityRestFilter.doHandleRequest(org.elasticsearch.security@8.10.2/SecurityRestFilter.java:96)
	at org.elasticsearch.xpack.security.rest.SecurityRestFilter.lambda$handleRequest$0(org.elasticsearch.security@8.10.2/SecurityRestFilter.java:87)
	at org.elasticsearch.xpack.security.rest.SecurityRestFilter$$Lambda$9277/0x00007effe9947160.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListener$2.onResponse(org.elasticsearch.server@8.10.2/ActionListener.java:177)
	at org.elasticsearch.xpack.security.authc.support.SecondaryAuthenticator.lambda$authenticateAndAttachToContext$3(org.elasticsearch.security@8.10.2/SecondaryAuthenticator.java:99)
	at org.elasticsearch.xpack.security.authc.support.SecondaryAuthenticator$$Lambda$9299/0x00007effe994e498.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(org.elasticsearch.server@8.10.2/ActionListenerImplementations.java:236)
	at org.elasticsearch.xpack.security.authc.support.SecondaryAuthenticator.authenticate(org.elasticsearch.security@8.10.2/SecondaryAuthenticator.java:109)
	at org.elasticsearch.xpack.security.authc.support.SecondaryAuthenticator.authenticateAndAttachToContext(org.elasticsearch.security@8.10.2/SecondaryAuthenticator.java:90)
	at org.elasticsearch.xpack.security.rest.SecurityRestFilter.handleRequest(org.elasticsearch.security@8.10.2/SecurityRestFilter.java:82)
	at org.elasticsearch.rest.RestController.dispatchRequest(org.elasticsearch.server@8.10.2/RestController.java:415)
	at org.elasticsearch.rest.RestController.tryAllHandlers(org.elasticsearch.server@8.10.2/RestController.java:544)
	at org.elasticsearch.rest.RestController.dispatchRequest(org.elasticsearch.server@8.10.2/RestController.java:317)
	at org.elasticsearch.http.AbstractHttpServerTransport.dispatchRequest(org.elasticsearch.server@8.10.2/AbstractHttpServerTransport.java:453)
	at org.elasticsearch.http.AbstractHttpServerTransport.handleIncomingRequest(org.elasticsearch.server@8.10.2/AbstractHttpServerTransport.java:549)
	at org.elasticsearch.http.AbstractHttpServerTransport.incomingRequest(org.elasticsearch.server@8.10.2/AbstractHttpServerTransport.java:426)
	at org.elasticsearch.http.netty4.Netty4HttpPipeliningHandler.handlePipelinedRequest(org.elasticsearch.transport.netty4@8.10.2/Netty4HttpPipeliningHandler.java:128)
	at org.elasticsearch.http.netty4.Netty4HttpPipeliningHandler.channelRead(org.elasticsearch.transport.netty4@8.10.2/Netty4HttpPipeliningHandler.java:118)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(io.netty.codec@4.1.94.Final/MessageToMessageDecoder.java:103)
	at io.netty.handler.codec.MessageToMessageCodec.channelRead(io.netty.codec@4.1.94.Final/MessageToMessageCodec.java:111)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(io.netty.codec@4.1.94.Final/MessageToMessageDecoder.java:103)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(io.netty.codec@4.1.94.Final/MessageToMessageDecoder.java:103)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:412)
	at org.elasticsearch.http.netty4.Netty4HttpHeaderValidator.forwardData(org.elasticsearch.transport.netty4@8.10.2/Netty4HttpHeaderValidator.java:194)
	at org.elasticsearch.http.netty4.Netty4HttpHeaderValidator.forwardFullRequest(org.elasticsearch.transport.netty4@8.10.2/Netty4HttpHeaderValidator.java:137)
	at org.elasticsearch.http.netty4.Netty4HttpHeaderValidator.lambda$requestStart$1(org.elasticsearch.transport.netty4@8.10.2/Netty4HttpHeaderValidator.java:120)
	at org.elasticsearch.http.netty4.Netty4HttpHeaderValidator$$Lambda$9237/0x00007effe99439a0.run(org.elasticsearch.transport.netty4@8.10.2/Unknown Source)
	at io.netty.util.concurrent.PromiseTask.runTask(io.netty.common@4.1.94.Final/PromiseTask.java:98)
	at io.netty.util.concurrent.PromiseTask.run(io.netty.common@4.1.94.Final/PromiseTask.java:106)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(io.netty.common@4.1.94.Final/AbstractEventExecutor.java:174)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(io.netty.common@4.1.94.Final/AbstractEventExecutor.java:167)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(io.netty.common@4.1.94.Final/SingleThreadEventExecutor.java:470)
	at io.netty.channel.nio.NioEventLoop.run(io.netty.transport@4.1.94.Final/NioEventLoop.java:566)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(io.netty.common@4.1.94.Final/SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(io.netty.common@4.1.94.Final/ThreadExecutorMap.java:74)
	at java.lang.Thread.runWith(java.base@20.0.2/Thread.java:1636)
	at java.lang.Thread.run(java.base@20.0.2/Thread.java:1623)



"elasticsearch[instance-0000000005][transport_worker][T#7]" #82 [330] daemon prio=5 os_prio=0 cpu=6962961.64ms elapsed=2312947.36s tid=0x00007eff8000ee70 nid=330 waiting for monitor entry  [0x00007effe20ec000]
   java.lang.Thread.State: BLOCKED (on object monitor)
	at net.shibboleth.utilities.java.support.component.AbstractInitializableComponent.isInitialized(net.shibboleth.utilities.java.support@8.4.0/AbstractInitializableComponent.java:43)
	- waiting to lock <0x000000062f76c4b8> (a org.elasticsearch.xpack.security.authc.saml.SamlRealm$PrivilegedHTTPMetadataResolver)
	at net.shibboleth.utilities.java.support.component.ComponentSupport.ifNotInitializedThrowUninitializedComponentException(net.shibboleth.utilities.java.support@8.4.0/ComponentSupport.java:104)
	at org.opensaml.saml.metadata.resolver.impl.AbstractMetadataResolver.resolveSingle(org.opensaml.saml.impl@4.3.0/AbstractMetadataResolver.java:262)
	at org.elasticsearch.xpack.security.authc.saml.SamlRealm.resolveEntityDescriptorWithPossibleRefresh(org.elasticsearch.security@8.10.2/SamlRealm.java:823)
	at org.elasticsearch.xpack.security.authc.saml.SamlRealm.resolveEntityDescriptor(org.elasticsearch.security@8.10.2/SamlRealm.java:801)
	at org.elasticsearch.xpack.security.authc.saml.SamlRealm.lambda$parseHttpMetadata$9(org.elasticsearch.security@8.10.2/SamlRealm.java:728)
	at org.elasticsearch.xpack.security.authc.saml.SamlRealm$$Lambda$4133/0x00007effe8f25920.run(org.elasticsearch.security@8.10.2/Unknown Source)
	at java.security.AccessController.executePrivileged(java.base@20.0.2/AccessController.java:809)
	at java.security.AccessController.doPrivileged(java.base@20.0.2/AccessController.java:571)
	at org.elasticsearch.xpack.security.authc.saml.SamlRealm.lambda$parseHttpMetadata$10(org.elasticsearch.security@8.10.2/SamlRealm.java:727)
	at org.elasticsearch.xpack.security.authc.saml.SamlRealm$$Lambda$4132/0x00007effe8f25708.get(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.xpack.security.authc.saml.SamlRealm.authenticate(org.elasticsearch.security@8.10.2/SamlRealm.java:541)
	at org.elasticsearch.xpack.security.authc.RealmsAuthenticator.lambda$consumeToken$4(org.elasticsearch.security@8.10.2/RealmsAuthenticator.java:147)
	at org.elasticsearch.xpack.security.authc.RealmsAuthenticator$$Lambda$9146/0x00007effe992f0d8.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.xpack.core.common.IteratingActionListener.onResponse(org.elasticsearch.xcore@8.10.2/IteratingActionListener.java:135)
	at org.elasticsearch.xpack.security.authc.RealmsAuthenticator.lambda$consumeToken$4(org.elasticsearch.security@8.10.2/RealmsAuthenticator.java:208)
	at org.elasticsearch.xpack.security.authc.RealmsAuthenticator$$Lambda$9146/0x00007effe992f0d8.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.xpack.core.common.IteratingActionListener.onResponse(org.elasticsearch.xcore@8.10.2/IteratingActionListener.java:135)
	at org.elasticsearch.xpack.security.authc.RealmsAuthenticator.lambda$consumeToken$4(org.elasticsearch.security@8.10.2/RealmsAuthenticator.java:208)
	at org.elasticsearch.xpack.security.authc.RealmsAuthenticator$$Lambda$9146/0x00007effe992f0d8.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.xpack.core.common.IteratingActionListener.onResponse(org.elasticsearch.xcore@8.10.2/IteratingActionListener.java:135)
	at org.elasticsearch.xpack.security.authc.RealmsAuthenticator.lambda$consumeToken$4(org.elasticsearch.security@8.10.2/RealmsAuthenticator.java:208)
	at org.elasticsearch.xpack.security.authc.RealmsAuthenticator$$Lambda$9146/0x00007effe992f0d8.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.xpack.core.common.IteratingActionListener.run(org.elasticsearch.xcore@8.10.2/IteratingActionListener.java:117)
	at org.elasticsearch.xpack.security.authc.RealmsAuthenticator.consumeToken(org.elasticsearch.security@8.10.2/RealmsAuthenticator.java:237)
	at org.elasticsearch.xpack.security.authc.RealmsAuthenticator.authenticate(org.elasticsearch.security@8.10.2/RealmsAuthenticator.java:83)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain.lambda$getAuthenticatorConsumer$5(org.elasticsearch.security@8.10.2/AuthenticatorChain.java:181)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain$$Lambda$9114/0x00007effe9928860.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.xpack.core.common.IteratingActionListener.onResponse(org.elasticsearch.xcore@8.10.2/IteratingActionListener.java:135)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain.lambda$getAuthenticatorConsumer$4(org.elasticsearch.security@8.10.2/AuthenticatorChain.java:189)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain$$Lambda$9141/0x00007effe992bce0.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListener$2.onResponse(org.elasticsearch.server@8.10.2/ActionListener.java:177)
	at org.elasticsearch.xpack.security.authc.ApiKeyAuthenticator.authenticate(org.elasticsearch.security@8.10.2/ApiKeyAuthenticator.java:50)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain.lambda$getAuthenticatorConsumer$5(org.elasticsearch.security@8.10.2/AuthenticatorChain.java:181)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain$$Lambda$9114/0x00007effe9928860.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.xpack.core.common.IteratingActionListener.onResponse(org.elasticsearch.xcore@8.10.2/IteratingActionListener.java:135)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain.lambda$getAuthenticatorConsumer$4(org.elasticsearch.security@8.10.2/AuthenticatorChain.java:189)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain$$Lambda$9141/0x00007effe992bce0.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListener$2.onResponse(org.elasticsearch.server@8.10.2/ActionListener.java:177)
	at org.elasticsearch.xpack.security.authc.OAuth2TokenAuthenticator.authenticate(org.elasticsearch.security@8.10.2/OAuth2TokenAuthenticator.java:44)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain.lambda$getAuthenticatorConsumer$5(org.elasticsearch.security@8.10.2/AuthenticatorChain.java:181)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain$$Lambda$9114/0x00007effe9928860.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.xpack.core.common.IteratingActionListener.onResponse(org.elasticsearch.xcore@8.10.2/IteratingActionListener.java:135)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain.lambda$getAuthenticatorConsumer$4(org.elasticsearch.security@8.10.2/AuthenticatorChain.java:189)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain$$Lambda$9141/0x00007effe992bce0.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListener$2.onResponse(org.elasticsearch.server@8.10.2/ActionListener.java:177)
	at org.elasticsearch.xpack.security.authc.ServiceAccountAuthenticator.authenticate(org.elasticsearch.security@8.10.2/ServiceAccountAuthenticator.java:49)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain.lambda$getAuthenticatorConsumer$5(org.elasticsearch.security@8.10.2/AuthenticatorChain.java:181)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain$$Lambda$9114/0x00007effe9928860.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.xpack.core.common.IteratingActionListener.run(org.elasticsearch.xcore@8.10.2/IteratingActionListener.java:117)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain.doAuthenticate(org.elasticsearch.security@8.10.2/AuthenticatorChain.java:137)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain.authenticateAsyncWithExistingAuthenticationToken(org.elasticsearch.security@8.10.2/AuthenticatorChain.java:109)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain.authenticateAsync(org.elasticsearch.security@8.10.2/AuthenticatorChain.java:82)
	at org.elasticsearch.xpack.security.authc.AuthenticationService.authenticate(org.elasticsearch.security@8.10.2/AuthenticationService.java:223)
	at org.elasticsearch.xpack.security.action.saml.TransportSamlAuthenticateAction.doExecute(org.elasticsearch.security@8.10.2/TransportSamlAuthenticateAction.java:65)
	at org.elasticsearch.xpack.security.action.saml.TransportSamlAuthenticateAction.doExecute(org.elasticsearch.security@8.10.2/TransportSamlAuthenticateAction.java:35)
	at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(org.elasticsearch.server@8.10.2/TransportAction.java:86)
	at org.elasticsearch.action.support.ActionFilter$Simple.apply(org.elasticsearch.server@8.10.2/ActionFilter.java:53)
	at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(org.elasticsearch.server@8.10.2/TransportAction.java:84)
	at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter.lambda$applyInternal$3(org.elasticsearch.security@8.10.2/SecurityActionFilter.java:163)
	at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter$$Lambda$7545/0x00007effe96bf228.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListenerImplementations$DelegatingFailureActionListener.onResponse(org.elasticsearch.server@8.10.2/ActionListenerImplementations.java:212)
	at org.elasticsearch.xpack.security.authz.AuthorizationService.lambda$authorizeAction$8(org.elasticsearch.security@8.10.2/AuthorizationService.java:455)
	at org.elasticsearch.xpack.security.authz.AuthorizationService$$Lambda$7664/0x00007effe96de770.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.xpack.security.authz.AuthorizationService$AuthorizationResultListener.onResponse(org.elasticsearch.security@8.10.2/AuthorizationService.java:1015)
	at org.elasticsearch.xpack.security.authz.AuthorizationService$AuthorizationResultListener.onResponse(org.elasticsearch.security@8.10.2/AuthorizationService.java:981)
	at org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(org.elasticsearch.server@8.10.2/ContextPreservingActionListener.java:32)
	at org.elasticsearch.xpack.security.authz.AuthorizationService.lambda$authorizeAction$9(org.elasticsearch.security@8.10.2/AuthorizationService.java:469)
	at org.elasticsearch.xpack.security.authz.AuthorizationService$$Lambda$7672/0x00007effe96debc0.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(org.elasticsearch.server@8.10.2/ActionListenerImplementations.java:236)
	at org.elasticsearch.xpack.security.authz.RBACEngine.authorizeClusterAction(org.elasticsearch.security@8.10.2/RBACEngine.java:185)
	at org.elasticsearch.xpack.security.authz.AuthorizationService.authorizeAction(org.elasticsearch.security@8.10.2/AuthorizationService.java:459)
	at org.elasticsearch.xpack.security.authz.AuthorizationService.maybeAuthorizeRunAs(org.elasticsearch.security@8.10.2/AuthorizationService.java:435)
	at org.elasticsearch.xpack.security.authz.AuthorizationService.lambda$authorize$3(org.elasticsearch.security@8.10.2/AuthorizationService.java:322)
	at org.elasticsearch.xpack.security.authz.AuthorizationService$$Lambda$7546/0x00007effe96bf450.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListener$2.onResponse(org.elasticsearch.server@8.10.2/ActionListener.java:177)
	at org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(org.elasticsearch.server@8.10.2/ContextPreservingActionListener.java:32)
	at org.elasticsearch.xpack.security.authz.RBACEngine.lambda$resolveAuthorizationInfo$0(org.elasticsearch.security@8.10.2/RBACEngine.java:150)
	at org.elasticsearch.xpack.security.authz.RBACEngine$$Lambda$7548/0x00007effe96bf890.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(org.elasticsearch.server@8.10.2/ActionListenerImplementations.java:236)
	at org.elasticsearch.xpack.security.authz.store.CompositeRolesStore.lambda$getRoles$4(org.elasticsearch.security@8.10.2/CompositeRolesStore.java:194)
	at org.elasticsearch.xpack.security.authz.store.CompositeRolesStore$$Lambda$7549/0x00007effe96bfaa0.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(org.elasticsearch.server@8.10.2/ActionListenerImplementations.java:236)
	at org.elasticsearch.xpack.security.authz.store.CompositeRolesStore.lambda$getRole$5(org.elasticsearch.security@8.10.2/CompositeRolesStore.java:212)
	at org.elasticsearch.xpack.security.authz.store.CompositeRolesStore$$Lambda$8014/0x00007effe9706c98.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(org.elasticsearch.server@8.10.2/ActionListenerImplementations.java:236)
	at org.elasticsearch.xpack.core.security.authz.store.RoleReferenceIntersection.lambda$buildRole$0(org.elasticsearch.xcore@8.10.2/RoleReferenceIntersection.java:49)
	at org.elasticsearch.xpack.core.security.authz.store.RoleReferenceIntersection$$Lambda$8015/0x00007effe9755c40.accept(org.elasticsearch.xcore@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(org.elasticsearch.server@8.10.2/ActionListenerImplementations.java:236)
	at org.elasticsearch.action.support.GroupedActionListener.onResponse(org.elasticsearch.server@8.10.2/GroupedActionListener.java:56)
	at org.elasticsearch.xpack.security.authz.store.CompositeRolesStore.buildRoleFromRoleReference(org.elasticsearch.security@8.10.2/CompositeRolesStore.java:292)
	at org.elasticsearch.xpack.security.authz.store.CompositeRolesStore$$Lambda$8013/0x00007effe9706a70.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.xpack.core.security.authz.store.RoleReferenceIntersection.lambda$buildRole$1(org.elasticsearch.xcore@8.10.2/RoleReferenceIntersection.java:53)
	at org.elasticsearch.xpack.core.security.authz.store.RoleReferenceIntersection$$Lambda$8016/0x00007effe9754800.accept(org.elasticsearch.xcore@8.10.2/Unknown Source)
	at java.lang.Iterable.forEach(java.base@20.0.2/Iterable.java:75)
	at org.elasticsearch.xpack.core.security.authz.store.RoleReferenceIntersection.buildRole(org.elasticsearch.xcore@8.10.2/RoleReferenceIntersection.java:53)
	at org.elasticsearch.xpack.security.authz.store.CompositeRolesStore.getRole(org.elasticsearch.security@8.10.2/CompositeRolesStore.java:210)
	at org.elasticsearch.xpack.security.authz.store.CompositeRolesStore.getRoles(org.elasticsearch.security@8.10.2/CompositeRolesStore.java:187)
	at org.elasticsearch.xpack.security.authz.RBACEngine.resolveAuthorizationInfo(org.elasticsearch.security@8.10.2/RBACEngine.java:146)
	at org.elasticsearch.xpack.security.authz.AuthorizationService.authorize(org.elasticsearch.security@8.10.2/AuthorizationService.java:338)
	at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter.lambda$applyInternal$4(org.elasticsearch.security@8.10.2/SecurityActionFilter.java:159)
	at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter$$Lambda$7544/0x00007effe96bf010.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(org.elasticsearch.server@8.10.2/ActionListenerImplementations.java:236)
	at org.elasticsearch.action.ActionListenerImplementations$MappedActionListener.onResponse(org.elasticsearch.server@8.10.2/ActionListenerImplementations.java:95)
	at org.elasticsearch.xpack.security.authc.AuthenticatorChain.authenticateAsync(org.elasticsearch.security@8.10.2/AuthenticatorChain.java:94)
	at org.elasticsearch.xpack.security.authc.AuthenticationService.authenticate(org.elasticsearch.security@8.10.2/AuthenticationService.java:261)
	at org.elasticsearch.xpack.security.authc.AuthenticationService.authenticate(org.elasticsearch.security@8.10.2/AuthenticationService.java:171)
	at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter.applyInternal(org.elasticsearch.security@8.10.2/SecurityActionFilter.java:155)
	at org.elasticsearch.xpack.security.action.filter.SecurityActionFilter.apply(org.elasticsearch.security@8.10.2/SecurityActionFilter.java:114)
	at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(org.elasticsearch.server@8.10.2/TransportAction.java:84)
	at org.elasticsearch.action.support.TransportAction.execute(org.elasticsearch.server@8.10.2/TransportAction.java:61)
	at org.elasticsearch.tasks.TaskManager.registerAndExecute(org.elasticsearch.server@8.10.2/TaskManager.java:199)
	at org.elasticsearch.client.internal.node.NodeClient.executeLocally(org.elasticsearch.server@8.10.2/NodeClient.java:113)
	at org.elasticsearch.client.internal.node.NodeClient.doExecute(org.elasticsearch.server@8.10.2/NodeClient.java:91)
	at org.elasticsearch.client.internal.support.AbstractClient.execute(org.elasticsearch.server@8.10.2/AbstractClient.java:379)
	at org.elasticsearch.action.ActionRequestBuilder.execute(org.elasticsearch.server@8.10.2/ActionRequestBuilder.java:59)
	at org.elasticsearch.xpack.security.rest.action.saml.RestSamlAuthenticateAction.lambda$innerPrepareRequest$0(org.elasticsearch.security@8.10.2/RestSamlAuthenticateAction.java:98)
	at org.elasticsearch.xpack.security.rest.action.saml.RestSamlAuthenticateAction$$Lambda$11473/0x00007effe9313cd0.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.rest.BaseRestHandler.handleRequest(org.elasticsearch.server@8.10.2/BaseRestHandler.java:103)
	at org.elasticsearch.xpack.security.rest.SecurityRestFilter.doHandleRequest(org.elasticsearch.security@8.10.2/SecurityRestFilter.java:96)
	at org.elasticsearch.xpack.security.rest.SecurityRestFilter.lambda$handleRequest$0(org.elasticsearch.security@8.10.2/SecurityRestFilter.java:87)
	at org.elasticsearch.xpack.security.rest.SecurityRestFilter$$Lambda$9277/0x00007effe9947160.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListener$2.onResponse(org.elasticsearch.server@8.10.2/ActionListener.java:177)
	at org.elasticsearch.xpack.security.authc.support.SecondaryAuthenticator.lambda$authenticateAndAttachToContext$3(org.elasticsearch.security@8.10.2/SecondaryAuthenticator.java:99)
	at org.elasticsearch.xpack.security.authc.support.SecondaryAuthenticator$$Lambda$9299/0x00007effe994e498.accept(org.elasticsearch.security@8.10.2/Unknown Source)
	at org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(org.elasticsearch.server@8.10.2/ActionListenerImplementations.java:236)
	at org.elasticsearch.xpack.security.authc.support.SecondaryAuthenticator.authenticate(org.elasticsearch.security@8.10.2/SecondaryAuthenticator.java:109)
	at org.elasticsearch.xpack.security.authc.support.SecondaryAuthenticator.authenticateAndAttachToContext(org.elasticsearch.security@8.10.2/SecondaryAuthenticator.java:90)
	at org.elasticsearch.xpack.security.rest.SecurityRestFilter.handleRequest(org.elasticsearch.security@8.10.2/SecurityRestFilter.java:82)
	at org.elasticsearch.rest.RestController.dispatchRequest(org.elasticsearch.server@8.10.2/RestController.java:415)
	at org.elasticsearch.rest.RestController.tryAllHandlers(org.elasticsearch.server@8.10.2/RestController.java:544)
	at org.elasticsearch.rest.RestController.dispatchRequest(org.elasticsearch.server@8.10.2/RestController.java:317)
	at org.elasticsearch.http.AbstractHttpServerTransport.dispatchRequest(org.elasticsearch.server@8.10.2/AbstractHttpServerTransport.java:453)
	at org.elasticsearch.http.AbstractHttpServerTransport.handleIncomingRequest(org.elasticsearch.server@8.10.2/AbstractHttpServerTransport.java:549)
	at org.elasticsearch.http.AbstractHttpServerTransport.incomingRequest(org.elasticsearch.server@8.10.2/AbstractHttpServerTransport.java:426)
	at org.elasticsearch.http.netty4.Netty4HttpPipeliningHandler.handlePipelinedRequest(org.elasticsearch.transport.netty4@8.10.2/Netty4HttpPipeliningHandler.java:128)
	at org.elasticsearch.http.netty4.Netty4HttpPipeliningHandler.channelRead(org.elasticsearch.transport.netty4@8.10.2/Netty4HttpPipeliningHandler.java:118)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(io.netty.codec@4.1.94.Final/MessageToMessageDecoder.java:103)
	at io.netty.handler.codec.MessageToMessageCodec.channelRead(io.netty.codec@4.1.94.Final/MessageToMessageCodec.java:111)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(io.netty.codec@4.1.94.Final/MessageToMessageDecoder.java:103)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(io.netty.codec@4.1.94.Final/MessageToMessageDecoder.java:103)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:412)
	at org.elasticsearch.http.netty4.Netty4HttpHeaderValidator.channelRead(org.elasticsearch.transport.netty4@8.10.2/Netty4HttpHeaderValidator.java:72)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(io.netty.codec@4.1.94.Final/ByteToMessageDecoder.java:346)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(io.netty.codec@4.1.94.Final/ByteToMessageDecoder.java:318)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(io.netty.codec@4.1.94.Final/MessageToMessageDecoder.java:103)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:412)
	at io.netty.handler.ssl.SslHandler.unwrap(io.netty.handler@4.1.94.Final/SslHandler.java:1383)
	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(io.netty.handler@4.1.94.Final/SslHandler.java:1246)
	at io.netty.handler.ssl.SslHandler.decode(io.netty.handler@4.1.94.Final/SslHandler.java:1295)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(io.netty.codec@4.1.94.Final/ByteToMessageDecoder.java:529)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(io.netty.codec@4.1.94.Final/ByteToMessageDecoder.java:468)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(io.netty.codec@4.1.94.Final/ByteToMessageDecoder.java:290)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:412)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(io.netty.transport@4.1.94.Final/DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:440)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(io.netty.transport@4.1.94.Final/AbstractChannelHandlerContext.java:420)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(io.netty.transport@4.1.94.Final/DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(io.netty.transport@4.1.94.Final/AbstractNioByteChannel.java:166)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(io.netty.transport@4.1.94.Final/NioEventLoop.java:788)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(io.netty.transport@4.1.94.Final/NioEventLoop.java:689)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(io.netty.transport@4.1.94.Final/NioEventLoop.java:652)
	at io.netty.channel.nio.NioEventLoop.run(io.netty.transport@4.1.94.Final/NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(io.netty.common@4.1.94.Final/SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(io.netty.common@4.1.94.Final/ThreadExecutorMap.java:74)
	at java.lang.Thread.runWith(java.base@20.0.2/Thread.java:1636)
	at java.lang.Thread.run(java.base@20.0.2/Thread.java:1623)

```

</details>

Relates to https://github.com/elastic/elasticsearch/issues/68004

**Comment by tvernum** [comment]
I think SAML authentication should just always run in a separate thread.
It is slow - there's signature checks, and potentially decryption as well - and it's done once via an explicit rest action, it's not part of the normal HTTP pipeline (where forking for cached credentials can be more costly than it's worth).

We could just make `TransportSamlAuthenticate` fork an explicit (generic?) thread pool.

**Comment by slobodanadamovic** [comment]
I believe we need to make all of the following actions execute on an explicit thread pool:

- `TransportSamlAuthenticateAction`
- `TransportSamlPrepareAuthenticationAction`
- `TransportSamlLogoutAction`
- `TransportSamlInvalidateSessionAction`

The `TransportSamlAuthenticateAction` and `TransportSamlPrepareAuthenticationAction` are both showing in the attached thread dump of this issue. 

The rest of the action can potentially having the same problem because they all execute the same code when trying to get the current `EntityDescriptor` by invoking the `idpDescriptor.get()`. The `EntityDescriptor` supplier will always execute the same code and try to [`resolveEntityDescriptorWithPossibleRefresh`](https://github.com/elastic/elasticsearch/blob/9dfd5dbd8f536518641c52b9068b061a5b7253c4/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlRealm.java#L801).

- `TransportSamlAuthenticateAction` -> `authenticate` -> [`this.idpDescriptor.get()`
](https://github.com/elastic/elasticsearch/blob/9dfd5dbd8f536518641c52b9068b061a5b7253c4/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlRealm.java#L541)
- `TransportSamlPrepareAuthenticationAction` -> `buildAuthenticationRequest` -> [`idpDescriptor.get()`](https://github.com/elastic/elasticsearch/blob/9dfd5dbd8f536518641c52b9068b061a5b7253c4/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlRealm.java#L893)
- `TransportSamlLogoutAction` -> `buildLogoutRequest` -> [`idpDescriptor.get()`](https://github.com/elastic/elasticsearch/blob/9dfd5dbd8f536518641c52b9068b061a5b7253c4/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlRealm.java#L914)
- `TransportSamlInvalidateSessionAction` -> `buildLogoutResponse` -> [`idpDescriptor.get()`](https://github.com/elastic/elasticsearch/blob/9dfd5dbd8f536518641c52b9068b061a5b7253c4/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlRealm.java#L935)

**Comment by DaveCTurner** [comment]
Sounds like a reasonable approach to me; just to add that I'd recommend liberal use of `org.elasticsearch.transport.Transports#assertNotTransportThread` and/or `org.elasticsearch.threadpool.ThreadPool#assertCurrentThreadPool` in the vicinity of any code that definitely shouldn't be on a transport thread. As well as catching bugs, it's helpful documentation for the reader.

**Comment by slobodanadamovic** [comment]
@DaveCTurner Thanks for confirming. I started with a straightforward approach which simply uses `GENERIC` thread pool (since adding a new thread pool for SAML seemed like not big enough reason) and  added `org.elasticsearch.transport.Transports#assertNotTransportThread` to assert that it's working as expected. But in my testing I seem to have hit the following bug: https://github.com/elastic/elasticsearch/issues/97916. I'm currently looking into ways to avoid it.

Here is a link to a draft PR: https://github.com/elastic/elasticsearch/pull/105232

**Comment by DaveCTurner** [comment]
If you grep the codebase for `97916` you'll find a reasonable workaround implemented in several other spots.
